### PR TITLE
Reader: Hook up site search to date ordering pill

### PR DESCRIPTION
**Pull request by blowery**
This teaches the site results and feed search to pass along a sort order to the API and stash different orderings under different search keys.

### CODE REVIEW

**Comment by blowery** [comment]
Currently based on #14939, once that lands, this will get rebased.

**Comment by gibrown** [comment]
Hmmm, not sure if i found a bug or what, but i searched for "news" on the main search, and got results. Then I went to following manage and searched for "news" and it doesn't work.

Also, just going to https://calypso.live/following/manage?q=data isn't getting any results

**Comment by blowery** [comment]
@gibrown should be fixed now

File: client/state/reader/feed-searches/reducer.js [hunk 1]
```diff
@@ -25,21 +25,21 @@ import { READER_FEED_SEARCH_RECEIVE } from 'state/action-types';
...
 
 /**
  * Tracks mappings between queries --> num results
  * Here is what the state tree may look like:
  * feedSearches: {
 		total: {
-			'wordpress tavern': 4,
-			'thingsldkjflskjfsdf': 0,
-			'chickens': 4000,
+			'wordpress tavern-F': 4,
```
Comment (bluefuton) [inline client/state/reader/feed-searches/reducer.js]: Can we explain what the `-U` and `-F` are here?

File: client/state/reader/feed-searches/query-key.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+/**
+ * Build a key for feed search queries
```
Comment (bluefuton) [inline client/state/reader/feed-searches/query-key.js]: Can we give some examples of query keys here, and what the various parts mean?

**Comment by gibrown** [comment]
LGTM, all seems to be working now.

File: client/reader/search-stream/site-results.jsx [hunk 3]
```diff
@@ -13,10 +13,14 @@ import { getReaderFeedsForQuery } from 'state/selectors';
...
 import { requestFeedSearch } from 'state/reader/feed-searches/actions';
 import ReaderInfiniteStream from 'components/reader-infinite-stream';
+import { SORT_BY_RELEVANCE, SORT_BY_LAST_UPDATED } from 'state/reader/feed-searches/actions';
+
+const pickSort = sort => ( sort === 'date' ? SORT_BY_LAST_UPDATED : SORT_BY_RELEVANCE );
 
 class SitesResults extends React.Component {
 	static propTypes = {
 		query: PropTypes.string,
+		sort: PropTypes.string,
```
Comment (samouri) [inline client/reader/search-stream/site-results.jsx]: PropTypes.oneOf( 'date', 'XXX' ); ?

**Comment by samouri** [comment]
I can confirm the ordering is different, but how can i tell its going by last_updated date?  We aren't currently showing the `last_updated` here. should we be?

File: client/reader/search-stream/site-results.jsx (hunk 3 above)
Comment (blowery) [inline client/reader/search-stream/site-results.jsx]: done

**Comment by blowery** [comment]
> I can confirm the ordering is different, but how can i tell its going by last_updated date? We aren't currently showing the last_updated here. should we be?

You can compare to the API results, to a degree. Try using feed search directly at https://developer.wordpress.com/docs/api/console/ (you want `v1/read/feed`)

**Comment by samouri** [comment]
Its not a one step process because feeds don't have dates on them in this api response format.  I confirmed that sort=last_updated is being sent out in the network request.  Anything else would be a backend change so I'll test that separately instead of blocking this PR.

**Pull request merged**